lib/index.js

```diff
@@ -0,0 +1,27 @@
+var _ = require('lodash');
+
+var permissions = [];
+
+function addPermissions(permission) {
+    if (permissions.indexOf(permission) === -1) {
+        permissions.push(permission);
+    }
+    return this;
+}
+module.exports = {
+    addPermissions: function () {
+        var params = arguments;
+        if (arguments.length === 1 && arguments[0] instanceof Array) {
+            params = arguments[0];
+        }
+        _.forEach(params, addPermissions);
```

Comment:
_.concat would be the fastest possible implementation of this - it takes the incoming permissions 'array' and merges it with the current permissions array v cheaply.

On the other hand, we may want to indicate somehow (throw an error, perhaps?) when we have duplicate keys, in which case you'd want to still use a backing associative array and check for the existence of the permission before you add it. Let me know which of the two approaches you decide upon.